specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.6-preview.1/keys.json

```diff
@@ -1410,6 +1458,58 @@
       },
       "description": "As of http://tools.ietf.org/html/draft-ietf-jose-json-web-key-18"
     },
+    "KeyAttributesWithAttestation": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/KeyAttributes"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "attestation": {
+          "$ref": "#/definitions/KeyAttestation",
+          "description": "The key or key version attestation."
+        }
+      },
+      "description": "The attributes of a key managed by the key vault service."
+    },
+    "KeyAttestation": {
+      "type": "object",
+      "properties": {
+        "certificates": {
+          "type": "string",
+          "format": "base64url",
+          "description": "The certificates used for validating attestation."
+        },
+        "firmwareVersion": {
+          "$ref": "#/definitions/FirmwareVersion",
+          "description": "The firmware version of the HSM."
+        },
+        "privateKey": {
+          "type": "string",
+          "format": "base64url",
+          "description": "The private key."
+        },
+        "publicKey": {
+          "type": "string",
+          "format": "base64url",
+          "description": "The public key in case of asymmetric keys."
+        }
+      }
+    },
+    "FirmwareVersion": {
```

Comment:
Why does this need to be a complex object? We use strings e.g., "major.minor" elsewhere - both in REST APIs as well as the SDKs. This is overkill, IMO.